Source/loadsave.cpp

```diff
@@ -2010,16 +2068,35 @@ void SaveGameData()
 		file.WriteBE<int32_t>(monstkill);
 
 	if (leveltype != DTYPE_TOWN) {
+		std::vector<int8_t> tmpAvailableMissiles(MaxMissilesForSaveGame);
+		size_t nextAvailableIndex = 0;
+
+		for (int8_t i = 0; i < MaxMissilesForSaveGame; i++) {
+			if (Missiles.find(i) == Missiles.end()) {
+				tmpAvailableMissiles[nextAvailableIndex] = i;
+				nextAvailableIndex += 1;
+			}
+		}
 		for (int monsterId : ActiveMonsters)
 			file.WriteBE<int32_t>(monsterId);
 		for (int i = 0; i < ActiveMonsterCount; i++)
 			SaveMonster(&file, Monsters[ActiveMonsters[i]]);
-		for (int missileId : ActiveMissiles)
-			file.WriteLE<int8_t>(missileId);
-		for (int missileId : AvailableMissiles)
+		size_t wroteMissiles = 0;
+		for (auto &missileWithId : Missiles) {
+			if (missileWithId.first < MaxMissilesForSaveGame) {
+				file.WriteLE<int8_t>(missileWithId.first);
+				wroteMissiles += 1;
+			}
+		}
+		for (; wroteMissiles < MaxMissilesForSaveGame; wroteMissiles++) {
+			file.WriteLE<int8_t>(0);
+		}

```

Comment:
```suggestion
		file.Skip<int8_t>(MaxMissilesForSaveGame - wroteMissiles);
```

Reply:
I needed to adjust `SaveHelper::Skip` (similar to `LoadHelper`), cause previous it didn't have a count parameter.